torchinductor/lowering.py

```diff
@@ -1321,7 +1321,9 @@ def full_like(x, fill_value, **kwargs):
 
 def tensor_constructor(fill_value):
     # torch.zeros, torch.ones, etc
-    def inner(*size, dtype=None, device=None, layout=0, pin_memory=False):
+    def inner(
+        *size, dtype=None, device=None, layout=0, pin_memory=False, memory_format=None
```

Comment:
Unused arg, but allows construction of tensors like `y = torch.empty((B, T, C), device='cuda', memory_format=torch.contiguous_format)`

Reply:
We should probably check that it's `contiguous`, otherwise we'd silently ignore things like `channels_last`?

Reply:
We don't use the value, do you want me to assert that it is contiguous_format?

Reply:
Yes, assert that it's either None or contiguous_format (similarly to how we do with pin_memory and layout)